pywbem_mock/_dmtf_cim_schema.py

```diff
@@ -0,0 +1,427 @@
+
+        For example::
+
+            cim_schema_2.49.0.mof
+        """
+        return self._schema_mof_filename
+
+    def __str__(self):
+        """
+        Return a short string representation of the
+        :class:`~pywbem_mock.DMTFCIMSchema` object for human consumption.
+        """
+
+        return '%s(schema_version=%s, schema_root_dir=%s, url=%s)' % \
+               (self.__class__.__name__, self.schema_version_str,
+                self.schema_root_dir, self._mof_zip_url)
+
+    def __repr__(self):
+        """
+        Return a string representation of the
+        :class:`~pywbem_mock.DMTFCIMSchema` object that is suitable for
+        debugging.
+        """
+        return '{0}(schema_version={1}, schema_root_dir={2}, ' \
+               'schema_mof_dir={3}, ' \
+               'schema_mof_filename={4})'.format(self.__class__.__name__,
+                                                 self.schema_version,
+                                                 self.schema_root_dir,
+                                                 self.schema_mof_dir,
+                                                 self.schema_mof_filename)
+
+    def _setup_dmtf_schema(self):
+        """
+        Install the DMTF CIM schema from the DMTF web site if it is not
+        already installed. This includes download from the DMTF of the
+        DMTF CIM schema zip file and expansion of that file into a subdirectory
+        defined by `schema_mof_dir`.
+
+        Once the schema zip file is downloaded into `schema_root_dir`, it is no
+        longer downloaded if this function is recalled since DMTF CIM Schema
+        releases are never modified; they are rereleased as new update versions.
+        If the schema zip file is in the schema directory, but no
+        'schema_mof_dir' subdirectory exists, the schema is unzipped without
+        downloading the zip file from the  DMTF.
+
+        This allows the DMTF CIM schema zip file to be downloaded once and
+        reused and the user to chose if they want to retain the expanded MOF
+        files or remove them with :meth:`~pywbem_mock.DMTFCIMSchema.clean` when
+        not being used.
+
+        If the schema is to be committed a source repository such as git
+        it is logical to commit only the DMTF CIM schema zip file. Creation of
+        the `schema_mof_dir` subdirectory will be created when the
+        :class:`pywbem_mock.DMTFCIMSchema` object is created.
+
+        Raises:
+            ValueError: If the schema cannot be retrieved from the DMTF web
+              site.
+            TypeError:  If the `schema_version` is not a valid tuple with 3
+              integer components
+        """
+        first = [True]  # List allows setting variable from inner function
+
+        def print_verbose(msg):
+            """
+            Inner method prints msg if var `first` is `False` and verbose is
+            `True`. Then it sets first to `False`.
```

Comment:
In this changed implementation, the method no longer prints the message on the first time (when `verbose` is `True`). This causes the first message to be lost, IMO.

I wonder why we need this logic with "first" at all. I understand it is nice when running under py.test, but the verbose printing of progress messages is a feature we make available to the user, so just always printing a line of text followed by newline should be acceptable.

Reply:
Removed all the first stuff and just print messages if verbose. much simpler for us old folks.

DONE